crates/astria-composer/src/searcher/collector.rs

```diff
@@ -102,22 +95,51 @@ impl Collector {
 
         use ethers::providers::Middleware as _;
         use futures::stream::StreamExt as _;
-        self.wait_for_geth(5, Duration::from_secs(5), 2.0)
-            .await
-            .wrap_err("failed connecting ")?;
 
         let Self {
             chain_id,
-            client,
             searcher_channel,
-            ..
+            status,
+            url,
         } = self;
 
+        let retry_config = tryhard::RetryFutureConfig::new(1024)
+            .exponential_backoff(Duration::from_millis(500))
+            .max_delay(Duration::from_secs(60))
+            .on_retry(
+                |attempt, next_delay: Option<Duration>, error: &ProviderError| {
+                    let error = error as &(dyn std::error::Error + 'static);
+                    let wait_duration = next_delay
+                        .map(humantime::format_duration)
+                        .map(tracing::field::display);
+                    warn!(
+                        attempt,
+                        wait_duration,
+                        error,
+                        "attempt to connect to geth node failed; retrying after backoff",
```

Comment:
```suggestion
                        "attempt to connect to eth node failed; retrying after backoff",
```

not sure why this was ever changed to geth lol

Reply:
It does make sense because we are using a geth specific RPC. I believe @itamarreif favored that.